src/main/resources/assets/js/components/AppListFilterComponent.jsx

```diff
@@ -0,0 +1,54 @@
+/** @jsx React.DOM */
+
+var React = require("react/addons");
+
+var AppListFilterComponent = React.createClass({
+  displayName: "AppListFilterComponent",
+
+  propTypes: {
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps: function () {
+    return {
+      value: "",
+      size: 0,
+      numFiltered: null
+    };
+  },
+
+  handleChange: function (event) {
+    this.props.onChange(event.target.value);
+  },
+
+  handleClear: function () {
+    this.props.value = "";
```

Comment:
Manipulating the props _inside_ the component isn't allowed. You should set a state here and working with states.
States are intended to use with changing data.